contracts/src/access/control.rs

```diff
@@ -99,20 +104,20 @@ pub enum Error {
     BadConfirmation(AccessControlBadConfirmation),
 }
 
-sol_storage! {
-    /// Information about a specific role.
-    pub struct RoleData {
-        /// Whether an account is member of a certain role.
-        mapping(address => bool) has_role;
-        /// The admin role for this role.
-        bytes32 admin_role;
-    }
+/// Information about a specific role.
+#[storage]
+pub struct RoleData {
```

Comment:
These changes to the way states are defined should probably be mentioned in the CHANGELOG.

Reply:
It also should produce exactly similar code by the end.. idk. Wdt @bidzyyys?

Reply:
Hmm is it needed to mention refactoring work in CHANGELOG if it does not noticeably affect users of our lib? It might seem like we're treating CHANGELOG as some poor type of git history :thinking:

Wdyt about documenting changes only if they have an actual effect on library users?

Reply:
@0xNeshi I agree

Reply:
I agree on `documenting changes only if they have an actual effect on library users`.